src/Cli.Tests/UtilsTests.cs

```diff
@@ -249,13 +249,22 @@ public void TestMergeConfig()
         fileSystem.AddFile("dab-config.Test.json", new MockFileData(ENV_BASED_CONFIG));
 
         RuntimeConfigLoader loader = new(fileSystem);
+        bool old = RuntimeConfigLoader.CheckPrecedenceForConfigInEngine;
```

Comment:
I lost our previous comment thread on this. We save the old value and set it to true for this specific test. However, if other tests change this flag without reverting the value back to how it was prior to test execution, does that mean those tests may not be deterministic?

Reply:
It's mostly a safety thing that we're catching it here, since it's a static value that we're working with the value will be impacted by whatever test was run before it, and we'll then impact all other tests with the following line.

Although, I thought I removed that property...

Reply:
I thought so too that you had removed this property in #1402. Maybe an incorrect merge?

Reply:
Yeah, something stuffed up with the merge I reckon, so I reverted it from the PR too.